Command/GenerateKeyPairCommand.php

```diff
@@ -0,0 +1,150 @@
+    }
+
+    protected function configure()
+    {
+        $this->setDescription('Generate public/private keys for use in your application.');
+        $this->addOption('dry-run', null, InputOption::VALUE_NONE, 'Do not update key files.');
+        $this->addOption('skip-if-exists', null, InputOption::VALUE_NONE, 'Do not update key files if they already exist.');
+        $this->addOption('overwrite', null, InputOption::VALUE_NONE, 'Overwrite key files if they already exist.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        list($privateKey, $publicKey) = self::generateKeyPair($this->passphrase);
+
+        $alreadyExists = $this->filesystem->exists($this->secretKey) || $this->filesystem->exists($this->publicKey);
+
+        if (true === $alreadyExists) {
+            try {
+                $this->handleExistingKeys($input);
+            } catch (\RuntimeException $e) {
+                if (0 === $e->getCode()) {
+                    $io->comment($e->getMessage());
+
+                    return 0;
+                }
+
+                $io->error($e->getMessage());
+
+                return 1;
+            }
+        }
+
+        if (true === $input->getOption('dry-run')) {
+            $io->success('Your keys have been generated!');
+            $io->newLine();
+            $io->writeln(sprintf('Update your private key in <info>%s</info>:', $this->secretKey));
+            $io->writeln($privateKey);
+            $io->newLine();
+            $io->writeln(sprintf('Update your public key in <info>%s</info>:', $this->publicKey));
+            $io->writeln($publicKey);
+
+            return 0;
+        }
+
+        if (!$io->confirm('You are about to replace your existing keys. Are you sure you wish to continue?')) {
+            $io->comment('Your action was canceled.');
+
+            return 0;
+        }
+
+        $this->filesystem->dumpFile($this->secretKey, $privateKey);
+        $this->filesystem->dumpFile($this->publicKey, $publicKey);
+
+        $io->success('Done!');
+
+        return 0;
+    }
+
+    private function handleExistingKeys(InputInterface $input): void
+    {
+        if (true === $input->getOption('skip-if-exists') && true === $input->getOption('overwrite')) {
+            throw new \RuntimeException('Both options `--skip-if-exists` and `--overwrite` cannot be combined.', 1);
+        }
+
+        if (true === $input->getOption('skip-if-exists')) {
+            throw new \RuntimeException('Your key files already exist, they won\'t be overriden.', 0);
+        }
+
+        if (false === $input->getOption('overwrite')) {
+            throw new \RuntimeException('Your keys already exist. Use the `--overwrite` option to force regeneration.', 1);
+        }
+    }
+
+    private static function generateKeyPair($passphrase)
+    {
+        $config = [
+            'digest_alg' => 'sha512',
```

Comment:
The following rules should be applied:

* RS256, RS384, RS512: minimum RSA key length is 2048 bytes
* ES256, ES384, ES512: curve are SEC P-256, P-384, **P-521** respectively
* EdDSA: only curve ed25519 is supported
* HS256, HS384, HS512: minimum length **shall** be 256, 384, 512 bytes respectively

Reply:
Seems legit, but this PR will take longer than expected 😄 
I'm not 100% familiar on how to translate this into openssl config args, but will do my best.

Reply:
For example, I can't find the corresponding curve for `ed25519` inside the [curve name list](https://www.php.net/manual/en/function.openssl-get-curve-names.php) - any ideas?

Reply:
You can gave a look at the source code of another library I wrote (https://github.com/web-token/jwt-framework/blob/v2.2/src/Component/KeyManagement/JWKFactory.php and https://github.com/web-token/jwt-framework/blob/v2.2/src/Component/Core/Util/ECKey.php#L138)

HSxxx key is just a random string.
Both RSxxx and ESxxx need openssl and the configuration is qui te simple.
EdDSA requires libsodium, but I am not sure it can be used here.

Reply:
Well, it looks like I managed to support all algorithms, except EdDSA, which is a little more particular (user will get an error in this case). I suggest not supporting EdDSA in key pair generation, as it seems a little more exotic to achieve, unless you insist.

Tests pass with all algorithms (see [latest commit](https://github.com/lexik/LexikJWTAuthenticationBundle/pull/817/commits/6744a05a09db60d92b777c4aae874d744d297416)), except I had to comment out the encryption/decryption test, since it only seems to work with RS* algorithms:

```
9) Lexik\Bundle\JWTAuthenticationBundle\Tests\Functional\Command\GenerateKeyPairCommandTest::testItGeneratesKeyPair with data set #14 ('HS512', 'dummy')
openssl_public_encrypt(): key type not supported in this PHP build!

/Users/benoit/Sandboxes/LexikJWTAuthenticationBundle/Tests/Functional/Command/GenerateKeyPairCommandTest.php:47

10) Lexik\Bundle\JWTAuthenticationBundle\Tests\Functional\Command\GenerateKeyPairCommandTest::testItGeneratesKeyPair with data set #15 ('ES256', 'dummy')
openssl_public_encrypt(): key type not supported in this PHP build!
```

Any ideas?

Reply:
OpenSSL key type for ESxxx should be `OPENSSL_KEYTYPE_EC`, not `OPENSSL_KEYTYPE_DSA`

Reply:
Fixed, thanks.